src/psm/src/pdnsim.cpp

```diff
@@ -209,6 +202,22 @@ int PDNSim::analyze_power_grid()
     _logger->report("######################################");
   }
 
+  std::map<odb::dbTechLayer*, std::map<odb::Point, double>> ir_drop;
+  std::vector<Node*> nodes       = gmat_obj->GetAllNodes();
+  int                vsize;
+  vsize = nodes.size();
+  odb::dbTech* tech        = _db->getTech();
+  for (int n = 0; n < vsize; n++) {
+    Node* node = nodes[n];
+    int node_layer_num = node->GetLayerNum();
+    NodeLoc node_loc = node->GetLoc();
+    odb::Point point = odb::Point(node_loc.first, node_loc.second);
+    double voltage = node->GetVoltage();
+    odb::dbTechLayer* node_layer = tech->findRoutingLayer(node_layer_num);
+    ir_drop[node_layer][point] = abs(irsolve_h->supply_voltage_src - voltage);
```

Comment:
Why is abs needed here?

Reply:
For GND nets. voltage is higher than supply which would be zero.